src/composeapp.cc

```diff
@@ -42,6 +44,55 @@ void ComposeApp::remove() {
   }
 }
 
+bool ComposeApp::check() {
+  bool result = false;
+
+  do {
+    bool cmd_res{false};
+    std::string cmd_output;
+
+    std::tie(cmd_res, cmd_output) = cmd("config");
+    if (!cmd_res) {
+      LOG_WARNING << "Missing and incomplete App config: " << name_;
+      break;
+    }
+
+    // calculate a number of container images that a given App consists of
+    std::string::size_type find_pos{0};
+    std::string::size_type line_pos{0};
+    int expected_container_number{0};
+
+    while ((find_pos = cmd_output.find("image:", line_pos)) != std::string::npos) {
+      if (cmd_output.substr(line_pos, (find_pos - line_pos)).find_first_of('#') == std::string::npos) {
+        ++expected_container_number;
+      }
+      line_pos = cmd_output.find('\n', find_pos);
+      ++find_pos;
+    }
+
+    // Get a number of running container images
+    std::tie(cmd_res, cmd_output) = cmd("ps -q");
+    if (!cmd_res) {
+      LOG_WARNING << "Failed to get a list of App's containers: " << name_;
+      break;
+    }
+
```

Comment:
I like this approach. Its what I was thinking also. I do think we should add a comment for future reference like:
// NOTE - This logic isn't foolproof. The services could *change* but the total remains the same. This logic does not detect that.

Reply:
I am not sure if I understand fully the comment. The services can change only via OTA if a new Target is created what triggers an update regardless of the given verification logic. Or do you mean that docker-compose.yml can be modified manually directly on a device?

I think it makes sense to add a comment saying that the given logic cannot detect all cases of App containers stopping (docker-compose stop).

Reply:
your comment idea is fine.

Reply:
ok